contracts/test/contract.SL2RD.test.js

```diff
@@ -41,6 +40,44 @@ contract("SL2RD", (accounts) => {
     );
   });
 
+  // assign roles for testing
+  const OWNER = accounts[0];
+  const PAYER = accounts[1];
+  const PAYEE = accounts[2];
+  const USDC_AMOUNT = web3.utils.toWei("10", "ether"); // 10 USDC
+
+  let sl2rdContract, usdcContract, shareContract, operatorRegistry;
+
+  before(async () => {
+    shareContract = await SHARE.new({ from: OWNER });
+    operatorRegistry = await OperatorRegistry.new({ from: OWNER });
+    // Disable code verification for tests
+    await shareContract.setCodeVerificationEnabled(false, { from: OWNER });
+  });
+
+  beforeEach(async () => {
+    // Deploy SL2RD contract
+    sl2rdContract = await SL2RD.new();
```

Comment:
@xiangformless Good progress! I think this `beforeEach` pattern with global variables declared using `let` is going to cause issues and may be related to your error. Please update the test to use the standard pattern that all other tests are using within the codebase, e.g. see [1] as a reference. These tests do not use module level global vars or the beforeEach pattern, instead they explicitly declare the values needed for the specific test case:

```javascript
  specify("Contract initialization", async () => {
    const shareContract = await SHARE.deployed();
    const operatorRegistry = await OperatorRegistry.deployed();
    const splitContract = await SL2RD_V2.new();
    await splitContract.initialize(
      "Unicorn Token",
      "UNICORN",
      10e6 /* totalShares */,
      10e3 /* totalPublicShares */,
      1000 /* batchSize */,
      shareContract.address,
      operatorRegistry.address,
      false /* testMode */
    );
    assert.equal(await splitContract.name(), "Unicorn Token");
    assert.equal(await splitContract.symbol(), "UNICORN");
    assert.equal(await splitContract.totalSupply(), 10e6);
    assert.equal(await splitContract.paymentBatchSize(), 1000);
    assert.equal(await splitContract.decimals(), 0);
    assert.equal(await splitContract.totalPublicShares(), 10e3);
    assert.equal(await splitContract.countPublicSharesDistributed(), 0);
    assert.equal(await splitContract.totalSlots(), 10e6);
    assert.equal(await splitContract.totalCommunitySlots(), 10e3);
    assert.equal(await splitContract.countAllocatedCommunitySlots(), 0);
  });
  ```
  
  Once you've made that change please let me know if you're encountering the same issue!

[1] https://github.com/formless-eng/share-v1-core/blob/main/contracts/test/contract.SL2RD_V2.test.js

Reply:
@xiangformless Can you paste the error?

Reply:
103 passing (8m)
  1 failing

  1) Contract: SL2RD
       receive method transfers USDC to the correct recipient:
     Error: Invalid number of parameters for "undefined". Got 4 expected 0!
      at Context.<anonymous> (contracts/test/contract.SL2RD.test.js:1090:47)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

Reply:
@xiangformless In these errors it's helpful to look specifically at the LOC referenced by the test. In this case LOC 1090 is:

```
const usdcContract = await MockERC20.new("Mock USDC", "USDC", 18, {
        from: OWNER,
});
```

However in the implementation of MockERC20 the constructor expects 0 arguments:

```
constructor() ERC20("Mock USDC", "USDC") {
        // Mint an initial supply of tokens for testing
        // Mint 1 million USDC tokens to deployer's address
        _mint(msg.sender, 1000000 * 10 ** decimals()); 
}
```

I was able to resolve by addressing that LOC, and the subsequent error I think you'll encounter is:

```
TypeError: usdcContract.mint is not a function 
```

This is because `mint` is actually a private function `_mint` which is already called within the constructor of `MockUSDC`. Rather than minting the USDC to the payer you should be able to transfer it.

Please let me know if this helps or if you have additional questions!

Reply:
thanks for your comments！the test is working now!